combat-trainer.lic

```diff
@@ -1175,7 +1175,17 @@ class TrainerProcess
     when 'App Careful'
       appraise(game_state, 'careful')
     when 'Tactics'
-      bput(%w(weave bob circle).sample, 'roundtime', 'There is nothing else', 'Face what', 'You must be closer', 'You must be standing', 'Strangely, you don\'t feel like fighting right now', 'flying too high for you to attack') unless game_state.npcs.empty?
+      if not game_state.npcs.empty?
+        case fput('analyze', 'roundtime', 'Analyze what', 'You reveal a massive','Your analysis reveals a massive', 'There is nothing else', 'Face what', 'You must be closer', 'You must be standing', 'Strangely, you don\'t feel like fighting right now', 'flying too high for you to attack') #unless 
+        when 'Analyze what'
+          fput('face next')
+          fput('analyze')
+        when 'You must be closer'
+          fput('engage')
+        when 'Your analysis reveals a massive','You reveal a massive'
```

Comment:
That's a really good question.  What this does is keeps looping 'analyze' until it sees a massive opening, then stops.  I'm not exactly sure *why* it works that way, but that's the desired effect and what I see when I run it